### PR TITLE
fix(TCK): `AccountCreateTransaction`

### LINKED ISSUE #1013: fix(TCK): JSON-RPC methods for `AccountCreateTransaction`
### Problem

Fix AccountCreate tck method to work with the latest version of the tck server

### Solution

-

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by 0xivanov**
**Description**:

Fix `AccountCreate` tck method to work with the latest version of the tck server

**Related issue(s)**:

Fixes #1013 #893

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: transaction.go [hunk 1]
```diff
@@ -4877,6 +4877,27 @@ func (tx *Transaction) execute(client *Client, e TransactionInterface) (Transact
 	}, nil
 }
 
+func (tx *Transaction) FreezeWith(client *Client, e TransactionInterface) (TransactionInterface, error) { //nolint
```
Comment (0xivanov) [inline transaction.go:4880]: TODO: we need a way of calling freeze on a transaction, without knowing the concrete type of the transaction. It's used in `FillOutTransaction` method

**Review by 0xivanov — COMMENTED** [review]
Adding some notes for reviewers and for myself.

File: crypto.go [hunk 2]
```diff
@@ -32,6 +32,7 @@ import (
 	"github.com/hashgraph/hedera-protobufs-go/services"
 	"github.com/pkg/errors"
 	"golang.org/x/crypto/pbkdf2"
+	protobuf "google.golang.org/protobuf/proto"
 )
 
 const _Ed25519PrivateKeyPrefix = "302e020100300506032b657004220420"
```
Comment (gsstoykov) [inline crypto.go:38]: Note for the future: This is not actually correct.

File: key_list.go [hunk 3]
```diff
@@ -73,7 +78,7 @@ func (kl *KeyList) AddAllPublicKeys(keys []PublicKey) *KeyList {
 }
 
 // String returns a string representation of the KeyList
```
Comment (gsstoykov) [inline key_list.go:80]: nit: Guess this is some old typo.

File: tck/cmd/server.go [hunk 4]
```diff
@@ -29,16 +29,10 @@ func main() {
...
-		"reset":                  errorHandlerWrapper(HandleError, handler.New(sdkService.Reset)),
-		"createAccount":          errorHandlerWrapper(HandleError, handler.New(accountService.CreateAccount)),
-		"createAccountFromAlias": errorHandlerWrapper(HandleError, handler.New(accountService.CreateAccountFromAlias)),
-		"getAccountInfo":         errorHandlerWrapper(HandleError, handler.New(accountService.GetAccountInfo)),
-		"updateAccountMemo":      errorHandlerWrapper(HandleError, handler.New(accountService.UpdateAccountMemo)),
-		"updateAccountKey":       errorHandlerWrapper(HandleError, handler.New(accountService.UpdateAccountKey)),
-		"deleteAccount":          errorHandlerWrapper(HandleError, handler.New(accountService.DeleteAccount)),
-		"generatePublicKey":      errorHandlerWrapper(HandleError, handler.New(methods.GeneratePublicKey)),
-		"generatePrivateKey":     errorHandlerWrapper(HandleError, handler.New(methods.GeneratePrivateKey)),
+		"setup":         errorHandlerWrapper(HandleError, handler.New(sdkService.Setup)),
```
Comment (gsstoykov) [inline tck/cmd/server.go:32]: We are binding the endpoints to `errorHandlerWrapper`. This doesn't look that good in my opinion. Would expect something like `setup: handler.New()`). Maybe we can rename to `HandleWrapper`.

File: transaction.go (hunk 1 above)
Comment (gsstoykov) [reply transaction.go:4880]: Could it be achievable if `FillOutTransaction` is a method of the Transactions themselves?

File: crypto.go (hunk 2 above)
Comment (0xivanov) [reply crypto.go:38]: We have the same in JS

File: key_list.go (hunk 3 above)
Comment (0xivanov) [reply key_list.go:80]: I don't see typo in the comment

File: transaction.go (hunk 1 above)
Comment (0xivanov) [reply transaction.go:4880]: I don't see the point, as `FillOutTransaction` would not fit in any way in the existing flows of the sdk. 
I just need a way to expose the freezeWith function, as it's exposed in Java.

File: crypto.go (hunk 2 above)
Comment (gsstoykov) [reply crypto.go:38]: I guess so but for example `30740201010420` would still be a valid prefix if you integrate the key data, curve data and etc. after it. Thats why I put in this notice.

File: key_list.go (hunk 3 above)
Comment (gsstoykov) [reply key_list.go:80]: Not a typo in the traditional sense but is String returns a string in the comment correct here?

File: transaction.go (hunk 1 above)
Comment (gsstoykov) [reply transaction.go:4880]: We can take a look. Maybe I don't have enough context.

File: key_list.go (hunk 3 above)
Comment (0xivanov) [reply key_list.go:80]: I think `String` is referring to the name of the method

**Review by 0xivanov — COMMENTED** [review]
Had to force push the whole tck server, because the `feat/tck-poc` was behind 80 commits and it was a mess.

File: tck/cmd/server.go (hunk 4 above)
Comment (0xivanov) [reply tck/cmd/server.go]: Renamed this to `postHandler`. Have a look.

File: tck/cmd/server.go [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+	bridge := jhttp.NewBridge(assigner, nil)
+
+	// Listen and redirect to bridge
+	http.HandleFunc("/", bridge.ServeHTTP)
+	port := os.Getenv("TCK_PORT")
+	if port == "" {
+		port = "80"
+	}
+	fmt.Println("Server is listening on port: " + port)
```
Comment (gsstoykov) [inline tck/cmd/server.go:46]: Maybe at some point we can refactor to using log or logrus.

File: tck/methods/key_test.go [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+		Threshold: 1,
+		Keys:      nil,
+	}
+
+	// When
+	_, err := GenerateKey(context.Background(), params)
+
+	// Then
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "invalid parameters: threshold should only be provided for thresholdKey types")
```
Comment (gsstoykov) [inline tck/methods/key_test.go:42]: Maybe we should think of a way to centralise the error messages. As a changes for the error in one place for example the `GenerateKey` may lead to changes in the tests and overall test inconsistency. I personally don't like checking against concrete strings.

File: tck/methods/sdk_test.go [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+	// When
+	response := sdkService.Reset(context.Background())
+
+	// Then
+	assert.Equal(t, "SUCCESS", response.Status)
+	assert.Nil(t, sdkService.Client)
+}
+
+func stringPointer(s string) *string {
```
Comment (gsstoykov) [inline tck/methods/sdk_test.go:70]: Would you need this if for example you had:
type SetupParams struct {
	OperatorAccountId  string  `json:"operatorAccountId"`
	OperatorPrivateKey string  `json:"operatorPrivateKey"`
	NodeIp             string `json:"nodeIp"`
	NodeAccountId      string `json:"nodeAccountId"`
	MirrorNetworkIp    string `json:"mirrorNetworkIp"`
}

File: tck/param/sdk.go [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package param
+
+type SetupParams struct {
```
Comment (gsstoykov) [inline tck/param/sdk.go:3]: Is there are reason for some of these being pointers and others not?

Comment (0xivanov) [reply tck/param/sdk.go:3]: Basically the ones that are being pointers are optional -> they can be nil.

Comment (0xivanov) [reply tck/param/sdk.go:3]: I will try to make every param struct be like this.

File: tck/methods/key_test.go (hunk 6 above)
Comment (0xivanov) [reply tck/methods/key_test.go]: Moved the errors to utils/key.go

File: tck/cmd/server.go (hunk 5 above)
Comment (0xivanov) [reply tck/cmd/server.go]: changed

File: tck/param/sdk.go (hunk 8 above)
Comment (0xivanov) [reply tck/param/sdk.go:3]: Done

File: tck/methods/sdk_test.go (hunk 7 above)
Comment (0xivanov) [reply tck/methods/sdk_test.go:70]: changed to use pointers where optional

**Review by gsstoykov — APPROVED** [review]
LGTM.

**Review by agadzhalov — APPROVED** [review]
LGTM

**Pull request merged**